### PR TITLE
feature(operator): First step to upgrade to Operator SDK 0.5.0

### PR DESCRIPTION
**Pull request by rhuss**
At least it compiles ;-) Still to check: Template handling (i.e. how to get the Config from the client) has probably been broken

### CODE REVIEW

**Comment by zregvart** [comment]
This will conflict with #4454

**Comment by rhuss** [comment]
@zregvart no problem, I will pick up your changes here.

File: install/operator/pkg/syndesis/action/initialize.go [hunk 1]
```diff
@@ -42,8 +47,8 @@ func (a *Initialize) Execute(syndesis *v1alpha1.Syndesis) error {
 		target.Status.Reason = v1alpha1.SyndesisStatusReasonMissing
 		target.Status.Description = ""
 		target.Status.Version = syndesisVersion
-		logrus.Info("Syndesis resource ", syndesis.Name, " initialized: installing version ", syndesisVersion)
+		a.log.Info("Syndesis resource initialized: installing version ", "name", syndesis.Name, "version", syndesisVersion)
 	}
 
-	return sdk.Update(target)
+	return cl.Update(context.TODO(), target)
```
Comment (astefanutti) [inline install/operator/pkg/syndesis/action/initialize.go]: The `Status()` endpoint should be used to update the status sub-resource.

File: install/operator/pkg/syndesis/action/checkupdates.go [hunk 2]
```diff
@@ -47,7 +53,7 @@ func (a *CheckUpdates) Execute(syndesis *v1alpha1.Syndesis) error {
 		target.Status.UpgradeAttempts = 0
 		target.Status.ForceUpgrade = false
 
-		logrus.Info("Starting upgrade of Syndesis resource ", syndesis.Name, " from version ", namespaceVersion, " to version ", a.operatorVersion)
-		return sdk.Update(target)
+		a.log.Info("Starting upgrade of Syndesis resource", "name", syndesis.Name, "currentVersion", namespaceVersion, "targetVersion", a.operatorVersion, "type", "checkUpdate")
+		return cl.Update(context.TODO(), target)
```
Comment (astefanutti) [inline install/operator/pkg/syndesis/action/checkupdates.go]: The `Status()` endpoint should be used to update the status sub-resource.

File: install/operator/pkg/syndesis/action/install.go [hunk 3]
```diff
@@ -94,21 +98,21 @@ func (a *Install) Execute(syndesis *v1alpha1.Syndesis) error {
 	target.Status.Reason = v1alpha1.SyndesisStatusReasonMissing
 	target.Status.Description = ""
 	addRouteAnnotation(target, syndesisRoute)
-	logrus.Info("Syndesis resource ", target.Name, " installed")
+	a.log.Info("Syndesis resource installed", "name", target.Name)
 
-	return sdk.Update(target)
+	return cl.Update(context.TODO(), target)
```
Comment (astefanutti) [inline install/operator/pkg/syndesis/action/install.go]: The `Status()` endpoint should be used to update the status sub-resource.

File: install/operator/pkg/syndesis/action/startup.go [hunk 4]
```diff
@@ -59,22 +63,22 @@ func (a *Startup) Execute(syndesis *v1alpha1.Syndesis) error {
 		target.Status.Phase = v1alpha1.SyndesisPhaseInstalled
 		target.Status.Reason = v1alpha1.SyndesisStatusReasonMissing
 		target.Status.Description = ""
-		logrus.Info("Syndesis resource ", syndesis.Name, " installed successfully")
-		return sdk.Update(target)
+		a.log.Info("Syndesis resource installed successfully", "name", syndesis.Name)
+		return cl.Update(context.TODO(), target)
```
Comment (astefanutti) [inline install/operator/pkg/syndesis/action/startup.go]: The `Status()` endpoint should be used to update the status sub-resource.

File: install/operator/pkg/syndesis/action/startup.go [hunk 5]
```diff
@@ -59,22 +63,22 @@ func (a *Startup) Execute(syndesis *v1alpha1.Syndesis) error {
...
 	} else if failedDeployment != nil {
 		target := syndesis.DeepCopy()
 		target.Status.Phase = v1alpha1.SyndesisPhaseStartupFailed
 		target.Status.Reason = v1alpha1.SyndesisStatusReasonDeploymentNotReady
 		target.Status.Description = "Some Syndesis deployments failed to startup within the allowed time frame"
-		logrus.Info("Startup failed for Syndesis resource ", syndesis.Name, ". Deployment ", *failedDeployment, " not ready")
-		return sdk.Update(target)
+		a.log.Info("Startup failed for Syndesis resource. Deployment not ready", "name", syndesis.Name, "deployment", *failedDeployment)
+		return cl.Update(context.TODO(), target)
```
Comment (astefanutti) [inline install/operator/pkg/syndesis/action/startup.go]: The `Status()` endpoint should be used to update the status sub-resource.

File: install/operator/pkg/syndesis/action/startup.go [hunk 6]
```diff
@@ -59,22 +63,22 @@ func (a *Startup) Execute(syndesis *v1alpha1.Syndesis) error {
...
 	} else {
 		target := syndesis.DeepCopy()
 		target.Status.Phase = v1alpha1.SyndesisPhaseStarting
 		target.Status.Reason = v1alpha1.SyndesisStatusReasonMissing
 		target.Status.Description = ""
-		logrus.Info("Waiting for Syndesis resource ", syndesis.Name, " to startup")
-		return sdk.Update(target)
+		a.log.Info("Waiting for Syndesis resource to startup", "name", syndesis.Name)
+		return cl.Update(context.TODO(), target)
```
Comment (astefanutti) [inline install/operator/pkg/syndesis/action/startup.go]: The `Status()` endpoint should be used to update the status sub-resource.

File: install/operator/pkg/syndesis/action/upgrade.go [hunk 7]
```diff
@@ -120,11 +126,11 @@ func (a *Upgrade) Execute(syndesis *v1alpha1.Syndesis) error {
 				target.Status.TargetVersion = targetVersion
 				target.Status.Description = "Upgrading from " + namespaceVersion + " to " + targetVersion + currentAttemptDescr
 
-				return sdk.Update(target)
+				return cl.Update(context.TODO(), target)
```
Comment (astefanutti) [inline install/operator/pkg/syndesis/action/upgrade.go]: The `Status()` endpoint should be used to update the status sub-resource.

File: install/operator/pkg/syndesis/action/upgrade.go [hunk 8]
```diff
@@ -135,20 +141,20 @@ func (a *Upgrade) Execute(syndesis *v1alpha1.Syndesis) error {
 			}
 			target.Status.UpgradeAttempts = target.Status.UpgradeAttempts + 1
 
-			return sdk.Update(target)
+			return cl.Update(context.TODO(), target)
```
Comment (astefanutti) [inline install/operator/pkg/syndesis/action/upgrade.go]: The `Status()` endpoint should be used to update the status sub-resource.

File: install/operator/pkg/syndesis/action/upgrade.go [hunk 9]
```diff
@@ -162,11 +168,11 @@ func completeUpgrade(syndesis *v1alpha1.Syndesis, newVersion string) error {
 	target.Status.UpgradeAttempts = 0
 	target.Status.ForceUpgrade = false
 
-	return sdk.Update(target)
+	return cl.Update(context.TODO(), target)
```
Comment (astefanutti) [inline install/operator/pkg/syndesis/action/upgrade.go]: The `Status()` endpoint should be used to update the status sub-resource.

File: install/operator/pkg/syndesis/action/upgradebackoff.go [hunk 10]
```diff
@@ -15,27 +16,36 @@ const (
...
+		a.log.Error(nil,"Upgrade of Syndesis resource failed too many times and will not be retried", "name", syndesis.Name)
 
 		target := syndesis.DeepCopy()
 		target.Status.Phase = v1alpha1.SyndesisPhaseUpgradeFailed
 		target.Status.Reason = v1alpha1.SyndesisStatusReasonTooManyUpgradeAttempts
 		target.Status.Description = "Upgrade failed too many times and will not be retried"
 		target.Status.ForceUpgrade = false
 
-		return sdk.Update(target)
+		return cl.Update(context.TODO(), target)
```
Comment (astefanutti) [inline install/operator/pkg/syndesis/action/upgradebackoff.go]: The `Status()` endpoint should be used to update the status sub-resource.

File: install/operator/pkg/syndesis/action/upgradebackoff.go [hunk 11]
```diff
@@ -74,10 +84,10 @@ func (a *UpgradeBackoff) Execute(syndesis *v1alpha1.Syndesis) error {
 		target.Status.Description = "Upgrading from " + currentVersion + " to " + targetVersion + " (attempt " + currentAttemptStr + ")"
 		target.Status.ForceUpgrade = true
 
-		return sdk.Update(target)
+		return cl.Update(context.TODO(), target)
```
Comment (astefanutti) [inline install/operator/pkg/syndesis/action/upgradebackoff.go]: The `Status()` endpoint should be used to update the status sub-resource.

File: install/operator/pkg/syndesis/action/upgradelegacy.go [hunk 12]
```diff
@@ -42,13 +50,13 @@ func (a *UpgradeLegacy) Execute(syndesis *v1alpha1.Syndesis) error {
 	target.Status.Description = ""
 	target.Status.Version = syndesisVersion
 
-	logrus.Info("Syndesis installation attached to resource ", syndesis.Name)
-	return sdk.Update(target)
+	a.log.Info("Syndesis installation attached to resource", "name", syndesis.Name)
+	return cl.Update(context.TODO(), target)
```
Comment (astefanutti) [inline install/operator/pkg/syndesis/action/upgradelegacy.go]: The `Status()` endpoint should be used to update the status sub-resource.

**Comment by rhuss** [comment]
@astefanutti thanks, I will take care of the reviews. Do you think it's easy to just upgraded to 0.5.0, too ? Is it just an updated of the dependency in Gopkg.toml ?

**Comment by astefanutti** [comment]
@rhuss yes I think it should be easy to jump bump directly to 0.5.0 into `Gopkg.toml`.

File: install/operator/deploy/crds/syndesis_v1alpha1_syndesis_crd.yaml [hunk 13]
```diff
@@ -0,0 +1,15 @@
+apiVersion: apiextensions.k8s.io/v1beta1
```
[the remark this answers is not in the document]
Comment (rhuss) [reply install/operator/deploy/crds/syndesis_v1alpha1_syndesis_crd.yaml]: @astefanutti sorry, I don't understand. You mean, there us a new API version for CustomResourceDefinition available ?

File: install/operator/Gopkg.toml [hunk 14]
```diff
@@ -1,18 +1,58 @@
...
+[[override]]
+  name = "sigs.k8s.io/controller-runtime"
+  version = "=v0.1.8"
 
 [[constraint]]
   name = "github.com/operator-framework/operator-sdk"
-  branch = "master"
-  # version = "=v0.0.5"
+  # The version rule is used for a specific release and the master branch for in between releases.
+  branch = "master" #osdk_branch_annotation
```
[the remark this answers is not in the document]
Comment (rhuss) [reply install/operator/Gopkg.toml]: 👍 setting it to =v0.5.0

File: install/operator/deploy/crds/syndesis_v1alpha1_syndesis_cr.yaml [hunk 15]
```diff
@@ -0,0 +1,7 @@
+apiVersion: syndesis.io/v1alpha1
+kind: Syndesis
+metadata:
+  name: example-syndesis
+spec:
+  # Add fields here
+  size: 3
```
[the remark this answers is not in the document]
Comment (rhuss) [reply install/operator/deploy/crds/syndesis_v1alpha1_syndesis_cr.yaml]: I'm going to remove that file, as we have a full example on level above (in "syndesis.yml". The CRD itself is defined in "../syndesis-crd.yml", so I will remove the whole directory here.

File: install/operator/deploy/crds/syndesis_v1alpha1_syndesis_crd.yaml (hunk 13 above)
[the remark this answers is not in the document]
Comment (astefanutti) [reply install/operator/deploy/crds/syndesis_v1alpha1_syndesis_crd.yaml]: @rhuss Sorry got confused with your question about `syndesis.io/v1alpha2`. Forget about it. Time for weekend too!

File: install/operator/deploy/crds/syndesis_v1alpha1_syndesis_cr.yaml (hunk 15 above)
[the remark this answers is not in the document]
Comment (astefanutti) [reply install/operator/deploy/crds/syndesis_v1alpha1_syndesis_cr.yaml]: Agreed, even better!

**Comment by rhuss** [comment]
@astefanutti everything should be fixed now.

File: install/operator/Gopkg.toml [hunk 16]
```diff
@@ -1,18 +1,58 @@
...
+  "k8s.io/code-generator/cmd/deepcopy-gen",
+  "k8s.io/code-generator/cmd/conversion-gen",
+  "k8s.io/code-generator/cmd/client-gen",
+  "k8s.io/code-generator/cmd/lister-gen",
+  "k8s.io/code-generator/cmd/informer-gen",
+  "k8s.io/code-generator/cmd/openapi-gen",
+  "k8s.io/gengo/args",
+]
+
+[[override]]
```
Comment (astefanutti) [inline install/operator/Gopkg.toml]: May be preferable to update the override to align with those of the operator SDK version 0.5.0.

**Comment by astefanutti** [comment]
@rhuss just a point related to the version overrides that could be updated to align with those of the operator SDK version 0.5.0. Otherwise LGTM.

**Comment by rhuss** [comment]
I will update the deps later with a later PR as I'm currently refactoring our action handler. I suggest to merge now, so that you can integrate your PR and I will rebase on top of this later.

Deal ?

**Comment by astefanutti** [comment]
Sounds good to me!

**Pull request merged**